#145 create a bonfhir Gluestack UI package

nickvallee opened the pull request:
linked to #145 

- Introduce Gluestack UI package and FhirValue

nickvallee commented on packages/gluestack-ui/src/r5/renderer.ts:
```diff
@@ -0,0 +1,8 @@
+
+export const GluestackUIRenderer: Partial<FhirUIRenderer> = {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  FhirValue: GlueStackFhirValue as any,
```
I will address the type stuff in later PRs because It was requested that I submit this PR as is.

julienblin replied:
I don't understand why this is `any` here. Just remove it and you should be fine.

julienblin commented on packages/gluestack-ui/src/r5/index.ts:
```diff
@@ -0,0 +1 @@
+export * from "./renderer";
```
Don't forget to also re-export anything in `./data-display` - to export the gluestackrendererprops as well.

julienblin commented on packages/gluestack-ui/src/r5/data-display/fhir-value.tsx:
```diff
@@ -0,0 +1,22 @@
+): ReactElement | null {
+  return (
+    <Text
+      className={props.className}
```
Does gluestack actually support className? Only when running in web, maybe?

nickvallee replied:
very good point. There was no error when testing in the ecpo app, but I will remove it

julienblin reviewed (changes requested):
Thanks @nickvallee for your contribution.

A couple of comments:
 - I see this is a new branch - what are you going to do with the `145-create-a-bonfhirgluestack-ui-package`?
  - please run the `pnpm build` prior to committing - this is missing some files in the `r4b` folder (which tells me this package was probably not tested...)
  - please cleanup the commit history and add a changeset - [see details here](https://bonfhir.dev/docs/contributing#changesets---versioning-packages) (now that we have a documentation...)
  - and finally, link the issue (#145) to the PR

nickvallee committed:
Fix #145 - Create a bonfhir Gluestack-ui package

nickvallee commented:
@julienblin documentation was really helpful thank you. 

As for the other branch, I will remove it now 👍 . All other changes have been done, I believe. ready for second pass of review.

Commit:
Add React as a peer dependency to gluestack-ui package.

This prevents react from being bundled as per the rollup script.

Commit:
Add React as a peer dependency to gluestack-ui package

This prevents react from being bundled as per the rollup script.
This also version the package as 0.1.0 to better reflect the actual status.

julienblin reviewed (approved):
Alright - this is good thank you!
I just pushed a small commit to add "react" as a peer dependency: https://github.com/bonfhir/bonfhir/pull/173/commits/ecb9fb0cb474609ced7b70ae43431901ef3d6f68

Without this, the react package was bundled in the final build (visible if you open the result of the build in the `dist` directory).

Please rebase and commit! The resulting package should be pushed to the npm registry automatically!

nickvallee commented:
@julienblin Thanks! Glad you caught that dependency issue. Back from vacation.

I have rebased the branch and committed the changes remotely.

> Please rebase and commit! The resulting package should be pushed to the npm registry automatically!

When you tell me to rebase and commit here, do you also mean merge the PR? I imagine NPM registry will only be pushed to the npm registry once we merge this PR into the main branch?

Just wanted to double check in case you prefer to be the one who merges the PRs. Happy to merge otherwise!

julienblin commented:
> @julienblin Thanks! Glad you caught that dependency issue. Back from vacation.
> 
> I have rebased the branch and committed the changes remotely.
> 
> > Please rebase and commit! The resulting package should be pushed to the npm registry automatically!
> 
> When you tell me to rebase and commit here, do you also mean merge the PR? I imagine NPM registry will only be pushed to the npm registry once we merge this PR into the main branch?
> 
> Just wanted to double check in case you prefer to be the one who merges the PRs. Happy to merge otherwise!

@nickvallee thanks - yeah go ahead and merge! Your assumption about packages and npm registry is mostly correct.
I tried to document the release process here: https://bonfhir.dev/docs/contributing#changesets---versioning-packages
